### PR TITLE
fix: update argument description for platform

### LINKED ISSUE #598: Improve --platform documentation to better explain its usage format
Current documentation:
```
$ oras manifest fetch -h
[Preview] Fetch manifest of the target artifact

** This command is in preview and under development. **

Example - Fetch raw manifest:
  oras manifest fetch localhost:5000/hello:latest

Example - Fetch the descriptor of a manifest:
  oras manifest fetch --descriptor localhost:5000/hello:latest

Example - Fetch manifest with specified media type:
  oras manifest fetch --media-type 'application/vnd.oci.image.manifest.v1+json' localhost:5000/hello:latest

Example - Fetch manifest with certain platform:
  oras manifest fetch --platform 'linux/arm/v5' localhost:5000/hello:latest

Example - Fetch manifest with prettified json result:
  oras manifest fetch --pretty localhost:5000/hello:latest

Usage:
  oras manifest fetch [flags] <name:tag|name@digest>

Aliases:
  fetch, get

Flags:
      --ca-file string                server certificate authority file for the remote registry
  -d, --debug                         debug mode
      --descriptor                    output the descriptor
  -h, --help                          help for fetch
      --insecure                      allow connections to SSL registry without certs
      --media-type strings            accepted media types
  -o, --output string                 output file path
  -p, --password string               registry password or identity token
      --password-stdin                read password or identity token from stdin
      --plain-http                    allow insecure connections to registry without SSL check
      --platform string               fetch the manifest of a specific platform if target is multi-platform capable
      --pretty                        prettify JSON objects printed to stdout
      --registry-config stringArray   auth config path
  -u, --username string               registry username
  -v, --verbose                       verbose output
```

Provides more examples or add the form messages for platform flag, e.g.,
```
--platform string               fetch the manifest of a specific platform if target is multi-platform capable, specifies the platform in the form os/arch[/variant][:osversion] (e.g. linux/amd64)
```

### PR DESCRIPTION
**Pull request by qweeah**
#Resolves #598

Signed-off-by: Billy Zha <jinzha1@microsoft.com>

### CODE REVIEW

File: cmd/oras/internal/option/platform.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ type Platform struct {
 
 // ApplyFlags applies flags to a command flag set.
 func (opts *Platform) ApplyFlags(fs *pflag.FlagSet) {
-	fs.StringVarP(&opts.Platform, "platform", "", "", "fetch the manifest of a specific platform if target is multi-platform capable")
+	fs.StringVarP(&opts.Platform, "platform", "", "", "only fetch a specific platform if target is multi-platform capable, in the form of os[/arch][/variant][:os_version])")
```
Comment (FeynmanZhou) [inline cmd/oras/internal/option/platform.go:34]: My suggestion: `only fetch a manifest with specifying a specific platform if the target is multi-platform capable, in the form of os[/arch][/variant][:os_version])"`

Comment (shizhMSFT) [inline cmd/oras/internal/option/platform.go:34]: If target is a manifest list, the specified platform is selected.
If target is a manifest, the platform is ensured to match the specified platform.

```suggestion
	fs.StringVarP(&opts.Platform, "platform", "", "", "ensure fetching a manifest of a platform in the form of os[/arch][/variant][:os_version])")
```

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by lizMSFT — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**